lib/domain/usecase/room/set_permission_level_interactor.dart

```diff
@@ -0,0 +1,30 @@
+import 'package:dartz/dartz.dart';
+import 'package:fluffychat/app_state/failure.dart';
+import 'package:fluffychat/app_state/success.dart';
+import 'package:fluffychat/domain/app_state/room/set_permission_level_state.dart';
+import 'package:matrix/matrix.dart';
+
+class SetPermissionLevelInteractor {
+  Stream<Either<Failure, Success>> execute({
+    required Map<User, int> userPermissionLevels,
+  }) async* {
+    try {
+      yield Right(SetPermissionLevelLoading());
+
+      for (final entry in userPermissionLevels.entries) {
+        final user = entry.key;
+        final level = entry.value;
+
+        await user.setPower(level);
+      }
+
+      yield const Right(SetPermissionLevelSuccess());
+    } catch (error) {
+      yield Left(
+        SetPermissionLevelFailure(
+          exception: error,
+        ),
+      );
```

Comment:
IMO, we should catch some common error and map it to dedicated Failure state to notice user like: NoPermissionFailure, ....